dockstore-integration-testing/src/test/java/io/dockstore/webservice/WebhookIT.java

```diff
@@ -726,22 +726,55 @@ public void testDockstoreYmlPublish() throws Exception {
+        assertEquals(3, version.getAuthors().size());
+        assertEquals(2, version.getOrcidAuthors().size());
+        Optional<Author> descriptorAuthor = version.getAuthors().stream().filter(author -> author.getName().equals("Descriptor Author")).findFirst();
+        assertTrue(descriptorAuthor.isPresent());
+        assertEquals("Descriptor Author", descriptorAuthor.get().getName());
+        assertEquals("Email should be set for descriptor author", "descriptor.author@gmail.com", descriptorAuthor.get().getEmail());
+
+        // Workflow containing only .dockstore.yml authors
+        client.handleGitHubRelease(authorsRepo, BasicIT.USER_2_USERNAME, "refs/heads/onlyDockstoreYmlAuthors", installationId);
+        workflow = client.getWorkflowByPath("github.com/" + authorsRepo + "/foobar", "versions,authors", BIOWORKFLOW);
+        version = workflow.getWorkflowVersions().stream().filter(v -> v.getName().equals("onlyDockstoreYmlAuthors")).findFirst().get();
+        assertEquals(2, version.getAuthors().size());
+        assertEquals(2, version.getOrcidAuthors().size());
+
+        // Workflow containing only a descriptor author
+        client.handleGitHubRelease(authorsRepo, BasicIT.USER_2_USERNAME, "refs/heads/onlyDescriptorAuthor", installationId);
+        workflow = client.getWorkflowByPath("github.com/" + authorsRepo + "/foobar", "versions,authors", BIOWORKFLOW);
+        version = workflow.getWorkflowVersions().stream().filter(v -> v.getName().equals("onlyDescriptorAuthor")).findFirst().get();
+        assertEquals(1, version.getAuthors().size());
+        assertEquals(0, version.getOrcidAuthors().size());
+
+        // Release workflow containing only a descriptor author again and test that it doesn't create a duplicate author
+        client.handleGitHubRelease(authorsRepo, BasicIT.USER_2_USERNAME, "refs/heads/onlyDescriptorAuthor", installationId);
+        workflow = client.getWorkflowByPath("github.com/" + authorsRepo + "/foobar", "versions,authors", BIOWORKFLOW);
+        version = workflow.getWorkflowVersions().stream().filter(v -> v.getName().equals("onlyDescriptorAuthor")).findFirst().get();
+        assertEquals(1, version.getAuthors().size());
+        assertEquals(0, version.getOrcidAuthors().size());
+
+        // Workflow containing multiple descriptor authors separated by a comma ("Author 1, Author 2") and no .dockstore.yml authors
+        client.handleGitHubRelease(authorsRepo, BasicIT.USER_2_USERNAME, "refs/heads/multipleDescriptorAuthors", installationId);
+        workflow = client.getWorkflowByPath("github.com/" + authorsRepo + "/foobar", "versions,authors", BIOWORKFLOW);
+        version = workflow.getWorkflowVersions().stream().filter(v -> v.getName().equals("multipleDescriptorAuthors")).findFirst().get();
+        assertEquals(1, version.getAuthors().size());
+        assertEquals(0, version.getOrcidAuthors().size());
```

Comment:
All the workflows are WDL, but there are some small changes to CWL below. 
Would be good to test/check out nextflow as well, I see a decent number of nextflow workflows with authors specified in search